packages/node/src/behaviors/on-off/OnOffServer.ts

```diff
@@ -216,15 +216,17 @@ export class OnOffBaseServer extends OnOffLogicBase {
         }
 
         if (transitionTime === 0) {
-            this.state.onOff = onOff;
-        } else {
-            this.internal.applyScenePendingOnOff = onOff;
-            this.internal.applySceneDelayTimer = Time.getTimer(
-                "delayed scene apply",
-                Millis(transitionTime),
-                this.callback(this.#applyDelayedSceneOnOffValue),
-            ).start();
+            if (onOff) {
+                return this.on();
+            }
+            return this.off();
         }
+        this.internal.applyScenePendingOnOff = onOff;
+        this.internal.applySceneDelayTimer = Time.getTimer(
+            "delayed scene apply",
+            Millis(transitionTime),
+            this.callback(this.#applyDelayedSceneOnOffValue),
```

Comment:
The type for the time callback is actually broken in that the `any` return type prevents detection of the fact that timers do not handle promises

So there's still a problem here I think.  Need to store the promise somewhere so it can be awaited when the behavior is closed, and handle errors

Simplest might be a Multiplex in this.internal.  More ambitious could be an "async timer" utility class that has an internal Multiplex and an async `close` method...  Useful for similar cases in the future.  Aaand fixing Timer.Callback return type

Reply:
updated with Multiplex